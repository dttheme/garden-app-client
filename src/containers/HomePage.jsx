import React, { Component } from 'react';
import { connect } from 'react-redux';
import WelcomeMessage from '../components/Welcome'
import mainHeaderImg from "../images/sprouts.jpg";


class Home extends Component {
  render() {
    return (
      <div
        className="mt7"
        >
        <WelcomeMessage />
      </div>
    );
  }
}

// const mapStateToProps = (state)
//
// const AppContainer = connect(
//   mapStateToProps,
// )

export default Home;
