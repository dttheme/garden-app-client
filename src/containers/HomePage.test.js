import React from 'react';
import { shallow, mount } from 'enzyme';
import { HomePage } from './HomePage';

describe('<HomePage />', () => {

  it('should render page', () => {
    const fakeAppState = {
      isFetchingUserInfo: true,
    };
    const fakeUser = {
      id: '123',
      username: 'j@e.com',
      firstName: 'Joe',
      isLoggedIn: true,
    };
    const wrapper = shallow(<HomePage user={fakeUser} appState={fakeAppState} fetchUserInfo={() => {}} />);
    expect(wrapper).toMatchSnapshot()
  });
});
