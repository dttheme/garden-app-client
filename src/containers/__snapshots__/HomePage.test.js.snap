// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage
    appState={
        Object {
            "isFetchingUserInfo": true,
          }
    }
    fetchUserInfo={[Function]}
    user={
        Object {
            "firstName": "Joe",
            "id": "123",
            "isLoggedIn": true,
            "username": "j@e.com",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="mt7"
>
        <ReturnMessage
                user={
                        Object {
                                "firstName": "Joe",
                                "id": "123",
                                "isLoggedIn": true,
                                "username": "j@e.com",
                              }
                }
        />
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ReturnMessage
          user={
                    Object {
                              "firstName": "Joe",
                              "id": "123",
                              "isLoggedIn": true,
                              "username": "j@e.com",
                            }
          }
/>,
        "className": "mt7",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "user": Object {
            "firstName": "Joe",
            "id": "123",
            "isLoggedIn": true,
            "username": "j@e.com",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="mt7"
>
          <ReturnMessage
                    user={
                              Object {
                                        "firstName": "Joe",
                                        "id": "123",
                                        "isLoggedIn": true,
                                        "username": "j@e.com",
                                      }
                    }
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ReturnMessage
            user={
                        Object {
                                    "firstName": "Joe",
                                    "id": "123",
                                    "isLoggedIn": true,
                                    "username": "j@e.com",
                                  }
            }
/>,
          "className": "mt7",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "user": Object {
              "firstName": "Joe",
              "id": "123",
              "isLoggedIn": true,
              "username": "j@e.com",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
